src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -778,31 +830,80 @@ private static uint HandleEvent(State state, ref StreamEvent evt)
 
         private static unsafe uint HandleEventRecv(State state, ref StreamEvent evt)
         {
-            StreamEventDataReceive receiveEvent = evt.Data.Receive;
-            for (int i = 0; i < receiveEvent.BufferCount; i++)
+            ref StreamEventDataReceive receiveEvent = ref evt.Data.Receive;
+
+            if (receiveEvent.BufferCount == 0)
             {
-                state.ReceiveQuicBuffers.Add(receiveEvent.Buffers[i]);
+                // This is a 0-length receive that happens once reads are finished (via abort or otherwise).
+                // State changes for this are handled elsewhere.
```

Comment:
Where?

Reply:
In PEER_SEND_SHUTDOWN / PEER_SEND_ABORT / SHUTDOWN_COMPLETE event handlers

Reply:
Can we put it in that comment instead of "elsewhere"? :smile: